server/src/main/java/org/gluu/oxtrust/api/RegistrationManagementService.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * oxAuth is available under the MIT License (2008). See http://opensource.org/licenses/MIT for full text.
+ *
+ * Copyright (c) 2014, Gluu
+ */
+
+package org.gluu.oxtrust.api;
+
+import com.wordnik.swagger.annotations.ApiOperation;
+import com.wordnik.swagger.annotations.ApiResponse;
+import com.wordnik.swagger.annotations.ApiResponses;
+import org.gluu.oxtrust.api.client.RegistrationManagementRequest;
+import org.gluu.oxtrust.api.client.RegistrationManagementResponse;
+import org.gluu.oxtrust.ldap.service.AttributeService;
+import org.gluu.oxtrust.ldap.service.JsonConfigurationService;
+import org.gluu.oxtrust.ldap.service.OrganizationService;
+import org.gluu.oxtrust.model.GluuOrganization;
+import org.gluu.oxtrust.model.RegistrationConfiguration;
+import org.gluu.oxtrust.service.filter.ProtectedApi;
+import org.gluu.oxtrust.util.OxTrustApiConstants;
+import org.gluu.oxtrust.util.OxTrustConstants;
+import org.slf4j.Logger;
+import org.xdi.config.oxtrust.AppConfiguration;
+import org.xdi.model.GluuAttribute;
+import org.xdi.util.StringHelper;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation for register REST web services.
+ *
+ * @author Shoeb Khan
+ * @version July 06, 2018
+ */
+@Path(OxTrustApiConstants.BASE_API_URL + "/configurations/registration")
+@ProtectedApi
+public class RegistrationManagementService {
+
+    public static final String SERVER_DENIED_THE_REQUEST = "Server denied the request.";
+    public static final String MALFORMED_REQUEST = "The request is malformed.";
+
+    @Inject
+    private Logger log;
+
+    @Inject
+    private OrganizationService organizationService;
+
+    @Inject
+    private AttributeService attributeService;
+
+    @Inject
+    private JsonConfigurationService jsonConfigurationService;
+
+
+    /**
+     * This operation retrieves the configuration info for the specified findAttributesForPattern criteria,
+     * or all the configurations if no findAttributesForPattern criteria specified.
+     * <p>
+     * Note:
+     * Due to some properties of {@link GluuAttribute} class, direct JSON mapping may result in JSON exceptions.
```

Comment:
I think this comment is irrelevant to this method, better to be removed.

Reply:
Right. It would more appropriate to have this in the method body. The comment is to stress why we are using JSON string and not the RegistrationManagementRequest object itself.